DesiMock

```diff
@@ -0,0 +1,195 @@
+        import json
+        from dla_cnn.data_loader import scan_flux_sample
+        from dla_cnn.localize_model import predictions_ann as predictions_ann_c2
+        from dla_cnn.data_loader import compute_peaks, get_lam_data
+        #from dla_cnn.data_loader import add_abs_to_sightline
+        from dla_cnn.absorption import add_abs_to_sightline
+        from dla_cnn.data_model.Prediction import Prediction
+        # Fluxes
+        fluxes = scan_flux_sample(self.flux, self.loglam, self.z_qso, -1, stride=1)[0]
+        # Model
+        with open(model_path+"_hyperparams.json",'r') as f:
+            hyperparameters = json.load(f)
+        loc_pred, loc_conf, offsets, density_data_flat = predictions_ann_c2(hyperparameters, fluxes, model_path)
+        self.prediction = Prediction(loc_pred=loc_pred, loc_conf=loc_conf, offsets=offsets, density_data=density_data_flat)
+        # Peaks
+        _ = compute_peaks(self)
+        # Absorbers?
+        add_abs_to_sightline(self)
+
+
+class DesiMock:
+    """
+    a class to load all spectrums from a Desi fits file.
+    :attribute wavelength array-like, save the wavelength of all spectrum (all spectrum have same wavelength array)
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux, 'ERROR':error, 'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas'information}
+      I think this format may be more readable and practical than simply take the data of each spectra apart.
+    """
+
+    def _init_(self, wavelength = None, data = {}):
+        self.wavelength = wavelength
+        self.data = data
+
+
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
+        :param truth_path: truth file path
+        :param zbest_path: zbest file path
+        :return: self.wavelength,self.data(contained all information we need)
+        """
+        spec = fits.open(spec_path)
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        self.wavelength = np.hstack((spec[2].data.copy(), spec[7].data.copy(), spec[12].data.copy()))
+
+        dlas_data = truth[3].data[truth[3].data.copy()['NHI']>19.3]
+        spec_dlas = {}
+        for item in dlas_data:
+            if item[2] not in spec_dlas:
+                spec_dlas[item[2]] = [('00'+str(item[3]-item[2]*1000),item[1],item[0])] # (dla_id, dla)
+            else:
+                spec_dlas[item[2]].append(('00'+str(item[3]-item[2]*1000),item[1],item[0]))
+
+        test = np.array([True if item in dlas_data['TARGETID'] else False for item in spec[1].data['TARGETID'].copy()])
+        for item in spec[1].data['TARGETID'].copy()[~test]:
+            spec_dlas[item] = []
+
+        spec_id = spec[1].data['TARGETID'].copy()
+        flux1 = spec[3].data.copy()
```

Comment:
Are these the 3 cameras?  If so, let's keep them separate.
We may want to only analyze the blue camera.

Reply:
Or add a flag array specfying which pixel belongs to which camera.